CMakeLists.txt

```diff
@@ -53,6 +53,8 @@ if(NOT CMAKE_CONFIGURATION_TYPES)
     set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE STRING "Choose the type of build." FORCE)
     # Set the possible values of build type for cmake-gui
     set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "RelWithDebInfo")
+  elseif(CMAKE_BUILD_TYPE MATCHES "^(None)$")
+    message(WARNING "Building with custom compiler flags is discouraged and can cause unexpected problems.")
```

Comment:
Setting CMAKE_BUILD_TYPE=None does not allow to Build with custom compile flags.